design-documents/scalars-for-templates.md

```diff
@@ -0,0 +1,154 @@
+### Use scalars and maps of scalars only as E-mail template variables
+#### The problem
+Developers can use any object and access any method of available objects when
+rendering an E-mail template, usually those objects would be models which methods
+can often alter database, file system and cache storages. Using those methods in
+templates can lead to unexpected behavior when during template rendering value will
+be written to database, files changed etc. That should not happen in presentation layer
+and E-mails are presentation. To make matters worse - administrators can make their own templates
+through admin area and thus being able to execute any code on the server.
+
+#### The solution
+To prevent these problems from happening we must force using prepared data in
+E-mail templates - scalars received by calling certain model layer services.
+Thus templates will only receive data without the possibility to execute any code.
+
+#### What needs to be done
+###### Allow access to arrays' values
```

Comment:
How would a content author iterate through an indexed array rather than an associative array? For instance, to render a list of order items?

Reply:
we have 'for' directive for that purpose